src/sim.rs

```diff
@@ -245,15 +245,31 @@ impl<'a> Sim<'a> {
                     }
                     is_finished = is_finished && handle.is_finished();
                 }
+
+                if let Role::Simulated {
+                    handle: Some(handle),
+                    ..
+                } = rt
+                {
+                    if handle.is_finished() {
+                        finished.push(addr);
+                    }
+                }
             }
 
             self.elapsed += tick;
 
-            // Check finished clients for err results. Runtimes are removed at
+            // Check finished clients and hosts for err results. Runtimes are removed at
             // this stage.
             for addr in finished.iter() {
-                if let Some(Role::Client { rt, handle }) = self.rts.remove(addr) {
-                    rt.block_on(handle)??;
+                if let Some(role) = self.rts.remove(addr) {
+                    let (rt, handle) = match role {
+                        Role::Client { rt, handle } => (rt, Some(handle)),
+                        Role::Simulated { rt, handle, .. } => (rt, handle),
+                    };
+                    if let Some(handle) = handle {
```

Comment:
I wonder if we want to log here to allow people to identify when their rt isn't shutting down gracefully? Like what task is blocking it etc.

Reply:
Perhaps! As it is, the tests that return results just print out the error, but not the source host/client and the backtrace does not go further than the test entry point.

Reply:
I think this will end up double logging. You'll see the turmoil warn if you have tracing enabled and then the test will render the error on failure. As an author of host software I expect my entry point to have proper error handling and logging so maybe we don't need to provide that here.